bin/problem.py

```diff
@@ -981,56 +981,69 @@ def validate_data(problem, mode: validate.Mode, constraints: dict | bool | None
 
     def validate_invalid_extra_data(p) -> bool:
         base_path = p.tmpdir / "invalid_data"
-        # find at least one (valid?) testcase to modify
-        test_paths = list(glob(p.path, "data/sample/**/*.in"))
-        test_paths += list(glob(p.path, "data/secret/**/*.in"))
-        test_paths = [path for path in test_paths if path.with_suffix(".ans").exists()]
-        test_paths.sort()
-        test_path = test_paths[0] if test_paths else None
-        if test_path:
-            test_case = test_path.relative_to(p.path / "data").with_suffix("")
-            log(f"Generating invalid testcases based on: {test_case}")
-        verbose(f"writing generated invalid testcases to: {base_path}")
-
-        # validator, dir, read, write,  copy, strict
-        validators: list[tuple[type[validate.AnyValidator], str, str, str, list[str], bool]] = [
-            (validate.InputValidator, "invalid_inputs", ".in", ".in", [], True),
-            (validate.AnswerValidator, "invalid_answers", ".ans", ".ans", [".in"], True),
-            (validate.OutputValidator, "invalid_outputs", ".ans", ".out", [".in", ".ans"], False),
+        # pick at most first 3 samples (assuming they are valid and have .ans)
+        samples: list[Path | Literal[None]] = list(glob(p.path, "data/sample/**/*.in"))
+        samples.sort()
+        samples = [path for path, _ in zip(samples, range(3))]
+        if not samples:
+            samples = [None]
```

Comment:
This can be simplified:
- The `sorted()` function automatically turns any iterable into a `list`
- To select the first three items, `zip` with `range(3)` can be replaced with `[:3]`
- I don't see a need to put `None` in the list, when any `sample is None` is ignored down in the loop below, so the list `testcases` will stay empty anyway :stuck_out_tongue: 

```suggestion
        samples: list[Path] = sorted(glob(p.path, "data/sample/**/*.in"))[:3]
```

Reply:
oh i thought `[:x]` could throw an error ^^
the None is important, and also the reason why you next suggestion does not work ^^

Reply:
Can you explain what the `None` does here? Perhaps also good to write that in comments because it feels a bit magical (at least to me, but perhaps also to future readers of the code) :stuck_out_tongue: Also, is it correct that `None` is now always added to the list, also when it has elements already?